fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1166,12 +1176,33 @@ struct RegisterClusterImpl {
 	}
 
 	ACTOR static Future<Void> run(RegisterClusterImpl* self) {
+		// Used if we need to rollback
+		state RemoveClusterImpl<DB> removeCluster(
+		    self->ctx.managementDb, self->clusterName, ClusterType::METACLUSTER_MANAGEMENT, ForceRemove::True, 5.0);
+
 		wait(self->ctx.runManagementTransaction(
 		    [self = self](Reference<typename DB::TransactionT> tr) { return registerInManagementCluster(self, tr); }));
 
 		// Don't use ctx to run this transaction because we have not set up the data cluster metadata on it and we
 		// don't have a metacluster registration on the data cluster
-		wait(configureDataCluster(self));
+		try {
+			wait(configureDataCluster(self));
+		} catch (Error& e) {
+			state Error error = e;
+			try {
+				// Attempt to unregister the cluster if we could not configure the data cluster. We should only do this
+				// if the data cluster state matches our ID and is in the REGISTERING in case somebody else has
+				// attempted to complete the registration or start a new one.
+				removeCluster.clusterId = self->clusterEntry.id;
+				removeCluster.legalClusterStates.insert(DataClusterState::REGISTERING);
+				wait(removeCluster.run());
+			} catch (Error& e) {
```

Comment:
I think we should add some trace events both after the successful case `wait(removeCluster.run());` and in this inner catch block so there are logs on whether the cluster removal succeeded